app/hook/usePost.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react"
+import { CardPost, Post } from "../constants/types/components_props/types"
+
+interface usePostParams {
+  yearFilter?: number
+  posts?: Post[]
+}
+export default function usePost({ yearFilter, posts }: usePostParams) {
+  const [cardPost, setCardPost] = useState<CardPost[]>([])
+  const [loading, setLoading] = useState<boolean>(true)
+
+  const mapPostToCardPost = (postsFromApi: Post[]): CardPost[] =>
+    postsFromApi.map((postFromApi) => {
+      const { descriptionContent } = postFromApi.contentPost[0]
+      return {
+        title: postFromApi.title,
+        id: postFromApi.id,
+        nameAuthor: postFromApi.nameAuthor,
+        data: postFromApi.data,
+        readingTime: postFromApi.readingTime,
+        category: postFromApi.category,
+        description: descriptionContent,
+      }
+    })
+  interface filterCardPostByYearParams {
```

Comment:
could we put this interface near the other interface? maybe after or before usePostParams?
for organization